apps/passport-server/src/routing/routes/telegramRoutes.ts

```diff
@@ -275,4 +280,35 @@ export function initTelegramRoutes(
       res.status(500).send(errorHtmlWithDetails(e as Error));
     }
   });
+
+  app.get("/telegram/user", async (req: Request, res: Response) => {
+    const authToken = req.headers.authorization;
+
+    try {
+      if (!process.env.API_AUTH_TOKEN) throw new Error(`No Auth token found`);
+      if (authToken !== `Bearer ${process.env.API_AUTH_TOKEN}`) {
+        return res.status(401).send("Unauthorized");
+      }
+
+      const telegramId = checkOptionalQueryParam(req, "telegramId");
+      const semaphoreId = checkOptionalQueryParam(req, "semaphoreId");
+      if (telegramId || semaphoreId) {
+        const db = await getDB();
+        if (telegramId) {
+          const user = await fetchUserByTelegramId(db, telegramId);
+          if (!user) throw new Error(`No user found`);
+          res.json(user);
+          //
+        } else if (semaphoreId) {
+          const user = await fetchUserBySemaphoreId(db, semaphoreId);
+          if (!user) throw new Error(`No user found`);
+          res.json(user);
+        }
+      }
+    } catch (e) {
```

Comment:
You don't need to have a try catch in the route - u caught errors are automatically caught, logged, and reported to rollbar / honeycomb.

Reply:
Fixed!